### DISCUSSION TITLE
Coloring issue on real-time point cloud streaming

### DISCUSSION THREAD

**Discussion by whizbuzzer** [isl-org/Open3D#3824] [Q&A] [answered]
Greetings,

I have been using Intel RealSense D455 for real-time point cloud streaming which I intend to use later on for other purposes. So far, I have been successful in creating the live stream as well as in adding coordinate axes to it. The output's color, however, appears to be muted/unsaturated:

<img width="641" alt="muted_pc" src="https://user-images.githubusercontent.com/58381051/127535499-b6a57ce6-f236-4fe3-9dde-8ab2bc8d1403.png">

Does anyone know how to fix this?

**Comment by ssheorey** [comment]
Can you share the code that you are using for this, or at least the camera and image processing portions? Also the RGB image as a reference?

**Comment by whizbuzzer** [comment, marked as answer]
So, I solved the issue by changing the format in the "config.enable_stream()" function, that is,

I changed

`config.enable_stream(rs.stream.color, 960, 540, rs.format.bgr8, 30)`

to

`config.enable_stream(rs.stream.color, 960, 540, rs.format.rgb8, 30)`

**Comment by julienguegan** [comment]
@whizbuzzer Hi, I am trying to do the same : visualize streaming from my camera D415. I use `vis.add_geometry(cloud)`, `vis.update_geometry(new_cloud)`, `vis.poll_events()`,  `vis.update_renderer()`. I get the cloud does not update in the window ... 

Do I miss something ? Could you share a minimal working code ?

Thank you 👍

**Reply by whizbuzzer** [reply]
Sure! What I did was create a flag that indicated whether the geometry was added or not. Let's call it "geom_added". Then, I put all these commands in a while loop. The add_geometry() needed to be executed only once and the update_geometry() handled the pointcloud update. Also, you would need to instantiate a PointCloud object outside the while loop. The code would look something like this:

```
cloud = o3d.geometry.PointCloud()
geom_added = False

while True:

     # Code to create point cloud called pcd

     cloud.colors = pcd.colors
     cloud.points = pcd.points
     if geom_added == False:
          vis.add_geometry(cloud)
          geom_added = True
     vis.update_geometry(cloud)
     vis.poll_events()
     vis.update_renderer()
```
